livekit-agents/livekit/agents/multimodal/multimodal_agent.py

```diff
@@ -25,9 +25,66 @@
     "function_calls_collected",
     "function_calls_finished",
     "metrics_collected",
+    "input_speech_transcription_failed",
 ]
 
 
+@dataclass
+class InputTranscription:
+    item_id: str
+    """id of the item"""
+    transcript: str | None
+    """transcript of the input audio"""
+    error: str | None = None
+    """error message"""
+
+
+@dataclass
+class Content:
+    response_id: str
+    item_id: str
+    output_index: int
+    content_index: int
+    text: str
+    audio: list[rtc.AudioFrame]
+    text_stream: AsyncIterable[str]
+    audio_stream: AsyncIterable[rtc.AudioFrame]
+    content_type: Literal["text", "audio"]
+
+
+@dataclass
+class Capabilities:
+    supports_chat_ctx_manipulation: bool = False
```

Comment:
Can we also use Protocol for the fields?
(E.g, ideally OpenAI code should change)

Reply:
Updated code as per suggestions
- now there're only few changes are in openai subclass, I've merged InputTranscriptionCompleted and InputTranscriptionFailed into one class, 
- removed toolcalls from RealtimeContent, I coundn't find we  use that anywhere, I also couldn't find use of accumalted audio or text, should I remove them??
- moved  methods _handle_response_content_done and _truncate_conversation_item to openai, because it's relavent there